test/common.py

```diff
@@ -146,11 +146,26 @@ def decorated(self, *args, **kwargs):
   return decorated
 
 
-def flaky(note=''):
+def flaky(note='', retries=3):
   assert not callable(note)
+
   if EMTEST_SKIP_FLAKY:
     return unittest.skip(note)
-  return lambda f: f
+
+  def decorated(f):
+    @wraps(f)
+    def modified(*args, **kwargs):
+      for i in range(retries):
```

Comment:
We already have a mechanism for this, I think I remember - ? It retries each test once after a failure. That saved us a lot of flakes when we turned it on. Can we just bump that from 2 to 3 for tests marked "flaky" rather than add a new mechanism?

Reply:
Ah yes, search for "`automatically retrying`" in `test/common.py`.

Reply:
Looks like that one is only applicable to browser tests?

Reply:
Yes, that mechanism for only for browser tests and is not implemented as nicely (IMHO).    I can try to merge them as a followup is thats OK?

Reply:
The browser test mechanism is specific to browser tests and can't be used a decorator on arbitrary tests.

Reply:
I updated this to make it opt in and only enable it when we are running against PRs